### DISCUSSION TITLE
Padding in symbols

### DISCUSSION THREAD

**Discussion by dpetka2001** [folke/trouble.nvim#554] [Q&A] [answered]
I'm using LazyVim and tried the `kanagawa` colorscheme and Lualine symbols look like this
![2024-07-17_12-27](https://github.com/user-attachments/assets/d49a30de-be97-429a-9b42-bd6c103b5d8c)
I think Trouble inserts some kind of padding at the end of the text. Is there a configuration setting that allows Trouble to not add padding at the end of the text? Or what is the name of the highlight group that Trouble uses for padding that should be changed in `kanagawa` instead so that the background is uniform in color?

**Comment by folke** [comment]
highlight groups with backgrounds is a total mess in lualine.

In order to fix the bg, I run the generated statusline through [this function](https://github.com/folke/trouble.nvim/blob/main/lua/trouble/config/highlights.lua#L97)

Not sure why that doesn't fix it for you though.

Are you sure it's not one of the lualine border components that has an incorrect hl group?

**Reply by dpetka2001** [reply]
Well I've just tried with default LazyVim installation and only 
```lua
  {
    "LazyVim/LazyVim",
    opts = function(_, opts)
      -- opts.colorscheme = "catppuccin-macchiato"
      opts.colorscheme = "kanagawa-dragon"
    end,
  },
  {
    "rebelot/kanagawa.nvim",
    opts = {},
  },
```
And this still happens. Initially there are 2 black squares in the lualine Trouble component, but I remove and re-insert the component from [here](https://github.com/LazyVim/LazyVim/blob/d263cf5dd1daa3c6b0c07c6fe6cc21b1d39988bb/lua/lazyvim/plugins/ui.lua#L202-L215) in my personal config and just add in there `padding = { left = 1, right = 0 }` and the black square becomes just 1. That's what made me think that it's some kind of padding added from Trouble lualine component at the end of the text, because the one black square from padding from Lualine disappeared after i set `padding.right = 0`. Not sure what other info I could give you.

**Reply by dpetka2001** [reply]
![2024-07-17_14-36](https://github.com/user-attachments/assets/28166129-8c3a-478e-90c3-6782f9a6fd4c)

The first black square after `opts` is just one. At the end there is a double black square. That is supposed to be the end of the Trouble lualine component. If I re-implement the code for Trouble lualine component in my personal config with `padding.right = 0`, that double black square becomes a single black square. And in the middle single black square, there shouldn't really be anything Lualine border component from what I can understand. The Lualine border components should only take effect at the boundaries between the components themselves. The first single black square should be coming from the padding of Trouble that puts at the end of text I believe (of course I might be wrong).

**Reply by dpetka2001** [reply, marked as answer]
@folke It was the `lualine_c_inactive` background color. It was different on kanagawa, while on tokyonight both `lualine_c_inactive` and `lualine_c_normal` have the same background color. It was indeed about setting the right highlight group for the corresponding colorscheme.

**Reply by folke** [reply]
That's odd though. Is that group set by kanagawa itself?
Or by the kanagawa lualine theme?

**Reply by dpetka2001** [reply]
It's set by the kanagawa lualine theme I believe? It's set [here](https://github.com/rebelot/kanagawa.nvim/blob/master/lua/lualine/themes/kanagawa.lua). I just changed the colors in the palette to be the same.
